src/components/projects/projectId/heuristics/heuristicId/shared/HeuristicRunButtons.tsx

```diff
@@ -21,6 +21,7 @@ export default function HeuristicRunButtons(props: HeuristicRunButtonsProps) {
 
     useEffect(() => {
         setCanStartHeuristic(checkCanStartHeuristic());
+        if (!props.checkCanStartHeuristic) return;
```

Comment:
This is some fix?

Reply:
yes i noticed one bug while dev, so i fixed it, i will add it as part of the dep array